sstables/index_entry.hh

```diff
@@ -25,7 +25,11 @@
 
 namespace sstables {
 
-using use_caching = bool_class<struct use_caching_tag>;
+enum use_caching {
```

Comment:
if we don't intent to use the enumerator as integer or use them in the enclosing scope directly, probably we could use scoped enum like
```c++
enum class use_caching {
    none,
    local,
    global
};
```

as the unscoped enum exposes the numerators as a named constant to the enclosing scope, in other words, `none`, `local` and `global` are now exposed to the namespace of `sstables`, so as long as we are in this naming space, these constants can be used without qualifying the enum name, see https://godbolt.org/z/nYxME33EY

but the scoped enum does not offer this. so the enumerators do not pollute the enclosing scope.